src/java/org/apache/cassandra/streaming/StreamSession.java

```diff
@@ -1338,18 +1339,26 @@ public String toString()
                '}';
     }
 
-    public StringBuilder boundStacktrace(StackTraceElement[] stacktrace, int limit)
+    public static void boundStackTrace(Throwable e, int limit, HashSet<Throwable> visited)
```

Comment:
you have no return... so you are just walking throwable and mutating them which is not at all safe.